### ISSUE TITLE
Crash on NTP

### ISSUE THREAD

**Issue by SergeyZhukovsky** [brave/brave-browser#28089]
I found that crash on GPS on 1.47.x. We need to do a check here for null and `mActivity.get().isFinishing()`:
https://github.com/brave/brave-core/blob/1.47.x/android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java#L383

```
Exception java.lang.IllegalArgumentException: You cannot start a load for a destroyed activity
  at com.bumptech.glide.manager.RequestManagerRetriever.assertNotDestroyed (RequestManagerRetriever.java:348)
  at com.bumptech.glide.manager.RequestManagerRetriever.get (RequestManagerRetriever.java:148)
  at com.bumptech.glide.manager.RequestManagerRetriever.get (RequestManagerRetriever.java:181)
  at com.bumptech.glide.Glide.with (Glide.java:813)
  at org.chromium.chrome.browser.ntp.BraveNewTabPageLayout.setNtpRecyclerView (BraveNewTabPageLayout.java:383)
  at org.chromium.chrome.browser.ntp.BraveNewTabPageLayout$3.run (BraveNewTabPageLayout.java:369)
  at android.os.Handler.handleCallback (Handler.java:808)
  at android.os.Handler.dispatchMessage (Handler.java:101)
  at android.os.Looper.loop (Looper.java:166)
  at android.app.ActivityThread.main (ActivityThread.java:7529)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:245)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:921)
```

**Comment by kjozwiak** [comment]
The above requires `1.47.190` or higher for `1.47.x` verification 👍

**Comment by kjozwiak** [comment]
Moving this into `1.48.x` as we neve received another `C109` minor chromium bump that would have triggered another `1.47.x` release. We'll leave https://github.com/brave/brave-core/milestone/267 as there's code that landed in `1.47.x` but we'll close off the `1.47.x - Release #6` milestone and move all the issues within `1.48.x`.

The above can be verified with `1.48.156` for `1.48.x` as it's been merged into a `1.48.x` a while ago as per https://github.com/brave/brave-core/pull/16908.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 13` using the following build(s):

```
Brave | 1.48.155 Chromium: 110.0.5481.77 (Official Build) (32-bit)
--- | ---
Revision | 65ed616c6e8ee3fe0ad64fe83796c020644d42af-refs/branch-heads/5481@{#839}
OS | Android 13; Build/TQ1A.230105.002
```

* ensured that you can opt-in/enable `Brave News` via the NTP without any issues
* ensured that you can scroll through the news feed without any issues
* ensured that you can tap/click on cards within the feed without any issues
